t3f/variables.py

```diff
@@ -3,6 +3,13 @@
 from t3f.tensor_train import TensorTrain
 from t3f.tensor_train_batch import TensorTrainBatch
 
+eager_mode_supported = False
+try:
+    from tensorflow.python.eager import context
+    eager_mode_supported = True
+except ImportError:
+    pass
```

Comment:
Maybe move all this logic to utils and create utils.in_eager_mode() that returns false if it's not supported?